File: app-java/src/main/java/com/java/bilibili/widget/ResizeLayout.java
package com.java.bilibili.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;

/**
 * Created by Eric.zhang on 2018/1/10.
 */

public class ResizeLayout extends RelativeLayout {
    private setParam a;

    public ResizeLayout(Context context) {
        super(context);
    }

    public ResizeLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ResizeLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public interface setParam {
        void a(int para1, int para2);
    }

    public void setOnSizeChangedListener(setParam param) {
        this.a = param;
    }

    @Override
    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
        super.onSizeChanged(w, h, oldw, oldh);
        if (a != null) {
            a.a(oldh, h);
        }
    }
}

File: app/src/main/java/com/kotlin/bilibili/widget/ResizeLayout.kt
package com.kotlin.bilibili.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout

class ResizeLayout : RelativeLayout {
    internal var a: setParam? = null

    constructor(context: Context) : super(context) {}
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
        if (this.a != null) {
            this.a!!.a(oldh, h)
        }
    }

    fun setOnSizeChangedListener(parama: setParam) {
        this.a = parama
    }

    interface setParam {
        fun a(paramInt1: Int, paramInt2: Int)
    }
}
